File: lib/Track.js
class Track {
  constructor(trackData, author) {
    this.title = trackData.title || trackData.videoDetails.title;

    this.url = trackData.url || null;

    this.author = trackData.author;

    this.requestedBy = author;

    this.author = trackData.channel
      ? trackData.channel.name
      : trackData.artist ||
        trackData.author.name ||
        trackData.videoDetails.author.name;
  }
}
module.exports = Track;

File: lib/Track.py
from discord import Embed, Color


class Track:
    def __init__(self, url, data, message):
        self.url = url
        self.stream_url = data.get("url") or None
        self.title = data.get("title")
        self.author = data.get("creator")
        self.duration = data.get("duration")
        self.cover = data.get("cover")
        self.requestedBy = message.author

    def get_embed(self):
        embed = Embed(title="Now Playing", description=self.title,
                      colour=Color.from_rgb(35, 219, 201))
        embed.set_footer(text="Requested by {0}".format(self.requestedBy.name))
        return embed
